### PR TITLE
(Bug Fix) Refresh page on submitting comment

### PR DESCRIPTION
**Pull request by christerforsgren91**
Pt: https://www.pivotaltracker.com/story/show/178339262

### User story
```
as a user
in order see my comment
i want the page to refresh when i submit
```
### Requested Changes
- Send comment with return key
- Fix page reload when submitting comment
- Display message if comment field is empty
- Clean up some code

### Video 
https://recordit.co/mbo30a3SYI
https://recordit.co/dnNmoOcjpb

### CODE REVIEW

File: src/components/layout/CommentInput.jsx [hunk 1]
```diff
@@ -18,11 +18,18 @@ const CommentInput = () => {
...
       setFocus(false);
+      refreshPage();
     } else {
       setMessage('Please subscribe to comment');
     }
   };
 
+  const refreshPage = () => {
+    setTimeout(() => {
+      window.location.reload();
```
Comment (tochman) [inline src/components/layout/CommentInput.jsx]: No. Update  state and have React do what it does bed. window.location.reload is a hack.

**Review by tochman — CHANGES_REQUESTED** [review]

File: src/components/layout/CommentInput.jsx (hunk 1 above)
Comment (1CIM) [reply src/components/layout/CommentInput.jsx]: Fixed

**Review by arunbhalli — APPROVED** [review]

File: src/components/layout/CommentInput.jsx [hunk 2]
```diff
@@ -8,28 +8,37 @@ const CommentInput = () => {
...
-        body: event.target.body.value,
+        body: input,
       };
-      Articles.createComment(comments, article.id);
-      setInput('');
-      setMessage('');
-      setFocus(false);
+      if (input.trim().length !== 0) {
+        Articles.createComment(comments, article.id);
+        setInput('');
```
Comment (LucasKnudsen) [inline src/components/layout/CommentInput.jsx:18]: Why the trim? :P

File: src/components/layout/CommentInput.jsx [hunk 3]
```diff
@@ -8,28 +8,37 @@ const CommentInput = () => {
...
       setMessage('Please subscribe to comment');
     }
   };
 
+  const submitCommentHandler = (event) => {
+    if (event.key === 'Enter') {
+      submitComment();
+    }
+  };
+
```
Comment (LucasKnudsen) [inline src/components/layout/CommentInput.jsx:34]: Did you get this to work? :) Was it the capital E?

File: src/components/layout/CommentInput.jsx [hunk 4]
```diff
@@ -40,14 +49,16 @@ const CommentInput = () => {
...
           {focus ? (
             <div className='comment-button-container'>
               <button
+                onClick={() => submitComment()}
                 className='submit-btn '
                 data-cy='comment-btn'
                 name='comment'
-                type='submit'>
+                type='button'>
                 Submit
```
Comment (LucasKnudsen) [inline src/components/layout/CommentInput.jsx]: Do you need to specify type button on a button element?

**Review by LucasKnudsen — APPROVED** [review]

File: src/components/layout/CommentInput.jsx (hunk 3 above)
Comment (1CIM) [reply src/components/layout/CommentInput.jsx:34]: Yeah E was the problem here 😅

File: src/components/layout/CommentInput.jsx (hunk 4 above)
Comment (1CIM) [reply src/components/layout/CommentInput.jsx]: Hmm we didn't try to not use any type, but the ` type='submit' ` fucked us hard for a while. it updated the site when we clicked the button but not enter and we couldn't figure it out why for a while then Christer changed the type to button and it worked so we decided not to touch that again for a while 😅

File: src/components/layout/CommentInput.jsx (hunk 2 above)
Comment (1CIM) [reply src/components/layout/CommentInput.jsx:18]: to get rig of the white spaces. so u can't write like 2 spaces in a row and not get the message me set for empty input :3

File: src/components/layout/CommentInput.jsx [hunk 5]
```diff
@@ -8,28 +8,37 @@ const CommentInput = () => {
...
+  };
+
   return (
     <div className='comment-outer-container'>
       <div className='comment-inner-container'>
         <h2 className='sign-up-message' data-cy='message'>
           {message}
         </h2>
-        <form onSubmit={(event) => submitComment(event)}>
+        <form>
```
Comment (tochman) [inline src/components/layout/CommentInput.jsx]: Not sure I understand why you keep the `form` at all...

File: src/components/layout/CommentInput.jsx [hunk 6]
```diff
@@ -40,14 +49,16 @@ const CommentInput = () => {
...
           />
           {focus ? (
             <div className='comment-button-container'>
               <button
+                onClick={() => submitComment()}
                 className='submit-btn '
                 data-cy='comment-btn'
                 name='comment'
-                type='submit'>
+                type='button'>
```
Comment (tochman) [inline src/components/layout/CommentInput.jsx]: A button is a button, and you don't NEED to give it a type... I'm sorry to hear that you spent a lot of time on this.

File: src/components/layout/CommentInput.jsx (hunk 5 above)
Comment (1CIM) [reply src/components/layout/CommentInput.jsx]: Yeah just said piss of to the form so now its gone 👍

File: src/components/layout/CommentInput.jsx (hunk 6 above)
Comment (1CIM) [reply src/components/layout/CommentInput.jsx]: removed

**Review by LucasKnudsen — APPROVED** [review]
Maybe today will be the day!

**Pull request merged**